lib/rpmds.c

```diff
@@ -1229,6 +1229,17 @@ static const struct rpmlibProvides_s rpmlibProvides[] = {
     { "rpmlib(FileDigests)", 		"4.6.0-1",
 	(		 RPMSENSE_EQUAL),
     N_("file digest algorithm is per package configurable") },
+#ifdef WITH_LIBGCRYPT
+    /*
+     * As rpmlib(FileDigestsGOST12) is available only when RPM is built with libcgrypt,
+     * to avoid other versions of RPM from misunderatanding hashes
+     * (see e.g. https://github.com/rpm-software-management/rpm/issues/959),
+     * require FileDigestsGOST12 separately
+     */
+    { "rpmlib(FileDigestsGOST12)", 		"4.16.0-1",
+	(		 RPMSENSE_RPMLIB|RPMSENSE_EQUAL),
+    N_("file digest can be GOST R 34.11 2012 (STREEBOG256, STREEBOG512)") },
```

Comment:
I'm not sure how to word this yet, but I don't like this wording much...

Reply:
Sorry, I did not understand, what do you mean?

Reply:
We don't track individual file digest algorithms for anything else either, lets keep it that way.
Rpm needs to better deal with unknown algorithms but rpmlib() dependencies are such a clumsy and ugly way...